shellscript/shellscript.go

```diff
@@ -3,13 +3,85 @@ package shellscript
 
 import (
 	"bufio"
+	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
 	"github.com/buildkite/shellwords"
+	"golang.org/x/exp/slices"
 )
 
+// Magic numbers for various types of executable files
+var (
```

Comment:
i'd like to thank the peanut gallery in the #serious-engineering channel for helping me with all of this binary nonsense